File: java/src/main/java/S002.java
public class S002 {

    private static final int LIMIT = 4 * 1000 * 1000;

    public static long solve() {
        long sum = 0;
        long prev = 1, cur = 2;
        while (cur < LIMIT) {
            if (cur % 2 == 0) {
                sum += cur;
            }
            long next = prev + cur;
            prev = cur;
            cur = next;
        }
        return sum;
    }
}

File: cpp/S002.cpp
#include <iostream>

using namespace std;

static const int LIMIT = 4 * 1000 * 1000;

int main() {
	int left = 1, right = 2;
	long sum = 0;
	while (right < LIMIT) {
		if (right % 2 == 0) {
			sum += right;
		}
		int next = left + right;
		left = right;
		right = next;
	}
	cout << sum << endl;
	return 0;
}
